src/ResourceManager/OperationalInsights/Commands.OperationalInsights/DataSources/GetAzureOperationalInsightsDataSourceCommand.cs

```diff
@@ -20,7 +20,7 @@
 
 namespace Microsoft.Azure.Commands.OperationalInsights
 {
-    [Cmdlet(VerbsCommon.Get, Constants.DataSource, DefaultParameterSetName = ByWorkspaceName), OutputType(typeof(List<PSDataSource>), typeof(PSDataSource))]
+    [Cmdlet(VerbsCommon.Get, Constants.DataSource, DefaultParameterSetName = ByWorkspaceNameByKind), OutputType(typeof(List<PSDataSource>), typeof(PSDataSource))]
```

Comment:
@haitch looking into this, this is going to be a breaking change. Existing scripts _could_ use `Get-AzureRmOperationalInsightsDataSource` with no parameters, even though it would do nothing (because the `ByWorkspaceName` parameter set isn't checked for in any if-statement in `ExecuteCmdlet`).

A problem arises now that you have changed the default parameter set to `ByWorkspaceNameByKind` because parameter `Kind` is mandatory in that parameter set, which means if you run `Get-AzureRmOperationalInsightsDataSource` with no parameters like before, you will now be prompted to enter a value for `Kind`, which will break that script that uses the cmdlet with no parameters.

My suggestion is to revert the change made to the default parameter set, write a warning to users who are using the `ByWorkspaceName` parameter set to let them know that this parameter set will be deprecated and changed to `ByWorkspaceNameByKind` in an upcoming release of Azure PowerShell, and then change the if-statements in `ExecuteCmdlet` to include `ByWorkspaceName`.